nixos/modules/tasks/filesystems.nix

```diff
@@ -321,39 +321,39 @@ in
 
   config = {
 
-    assertions = let
+    assertions.fileSystems = let
       ls = sep: concatMapStringsSep sep (x: x.mountPoint);
       resizableFSes = [
         "ext3"
         "ext4"
         "btrfs"
         "xfs"
       ];
-      notAutoResizable = fs: fs.autoResize && !(builtins.elem fs.fsType resizableFSes);
-    in [
-      { assertion = ! (fileSystems' ? cycle);
-        message = "The ‘fileSystems’ option can't be topologically sorted: mountpoint dependency path ${ls " -> " fileSystems'.cycle} loops to ${ls ", " fileSystems'.loops}";
-      }
-      { assertion = ! (any notAutoResizable fileSystems);
-        message = let
-          fs = head (filter notAutoResizable fileSystems);
-        in ''
+    in {
+      topologicallySorted = {
+        assertion = ! (fileSystems' ? cycle);
+        message = ''
+          The ‘fileSystems’ option can't be topologically sorted:
+          mountpoint dependency path ${ls " -> " fileSystems'.cycle or []} loops to ${ls ", " fileSystems'.loops or []}
+        '';
+      };
```

Comment:
For anyone interested, this is an example of an assertion that earlier relied on its error condition. It was expecting both `fileSystems'.cycle` and `fileSystems'.loops` to exist, implied by `fileSystems' ? cycle`.